### PR TITLE
extconf.rb - add logging for OpenSSL versions

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

This PR only changes logging when compiling with OpenSSL.

Currently, `extconf.rb` has comments about which functions exist in various OpenSSL versions.  This PR changes the comments to ouput logging.  Example below, all lines beginning with three dashes are new:
```
checking for pkg-config for openssl... [" ", "", "-lssl -lcrypto"]
──── Using OpenSSL pkgconfig (openssl.pc) ────
checking for openssl/bio.h... yes

──── Below are yes for 1.0.2 & later ────
checking for DTLS_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_session_cache_mode(NULL, 0) in openssl/ssl.h... yes
checking for SSL_CTX_set_ecdh_auto(NULL, 0) in openssl/ssl.h... yes

──── Below are yes for 1.1.0 & later ────
checking for TLS_server_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_min_proto_version(NULL, 0) in openssl/ssl.h... yes
checking for X509_STORE_up_ref()... yes

──── Below is yes for 1.1.0 and later, but isn't documented until 3.0.0 ────
checking for SSL_CTX_set_dh_auto(NULL, 0) in openssl/ssl.h... yes

──── Below is yes for 1.1.1 & later ────
checking for SSL_CTX_set_ciphersuites(NULL, "") in openssl/ssl.h... yes

──── Below is yes for 3.0.0 & later ────
checking for SSL_get1_peer_certificate() in openssl/ssl.h... yes
```

These comments appear in CI logs, and if one is installing the gem, the log fie is `gem_make.out` in the proper Ruby 'extensions' folder.

### Your checklist for this pull request
- [ ] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: ext/puma_http11/extconf.rb [hunk 1]
```diff
@@ -33,26 +33,27 @@
...
 
-    # below are yes for 1.1.0 & later
+    puts "\n──── Below are yes for 1.1.0 & later ────"
     have_func "TLS_server_method"                      , ssl_h
     have_func "SSL_CTX_set_min_proto_version(NULL, 0)" , ssl_h
-
-    # below are yes for 1.1.1 & later
-    have_func "SSL_CTX_set_ciphersuites(NULL, \"\")"   , ssl_h
-
     have_func "X509_STORE_up_ref"
```
Comment (dentarg) [inline ext/puma_http11/extconf.rb:44]: Should `X509_STORE_up_ref` be under 1.1.1? It moved here

https://www.openssl.org/docs/man1.1.0/man3/X509_STORE_up_ref.html is "Page Not Found" but https://www.openssl.org/docs/man1.1.1/man3/X509_STORE_up_ref.html exist

Comment (MSP-Greg) [reply ext/puma_http11/extconf.rb:44]: OpenSSL docs, especially older ones...

See https://github.com/openssl/openssl/blob/OpenSSL_1_1_0/crypto/x509/x509_lu.c#L220

Comment (MSP-Greg) [reply ext/puma_http11/extconf.rb:47]: I just updated the PR, added links for the two 'special items'...

**Comment by MSP-Greg** [comment]
@dentarg

I reviewed this again, and two of the `have_func` calls were no longer being used in `mini_ssl.c`, so the last commit removed them...

**Pull request merged**